modern/src/reports/TripReportPage.js

```diff
@@ -30,40 +30,93 @@ const TripReportPage = () => {
+  }, {
+    headerName: t('reportEndAddress'),
+    field: 'endAddress',
+    type: 'string',
+    hide: true,
+    flex: 1,
+  }, {
+    headerName: t('sharedDistance'),
+    field: 'distance',
+    type: 'number',
+    flex: 1,
+    valueFormatter: ({ value }) => formatDistance(value, distanceUnit),
+  }, {
+    headerName: t('reportAverageSpeed'),
+    field: 'averageSpeed',
+    type: 'number',
+    flex: 1,
+    valueFormatter: ({ value }) => formatSpeed(value, speedUnit),
+  }, {
+    headerName: t('reportMaximumSpeed'),
+    field: 'maxSpeed',
+    type: 'number',
+    flex: 1,
+    valueFormatter: ({ value }) => formatSpeed(value, speedUnit),
+  }, {
+    headerName: t('reportDuration'),
+    field: 'duration',
+    type: 'string',
+    flex: 1,
+    valueFormatter: ({ value }) => formatHours(value),
+  }, {
+    headerName: t('reportSpentFuel'),
+    field: 'spentFuel',
+    type: 'number',
+    flex: 1,
+    hide: true,
+    valueFormatter: ({ value }) => formatVolume(value, volumeUnit),
+  }, {
+    headerName: t('sharedDriver'),
+    field: 'driverName',
+    type: 'string',
+    flex: 1,
+    hide: true                                        
+  }]
+
   return (
     <ReportLayoutPage filter={<Filter setItems={setItems} />}>
-      <TableContainer component={Paper}>
-        <Table>
-          <TableHead>
-            <TableRow>
-              <TableCell>{t('reportStartTime')}</TableCell>
-              <TableCell>{t('reportStartOdometer')}</TableCell>
-              <TableCell>{t('reportEndTime')}</TableCell>
-              <TableCell>{t('reportEndOdometer')}</TableCell>
-              <TableCell>{t('sharedDistance')}</TableCell>
-              <TableCell>{t('reportAverageSpeed')}</TableCell>
-              <TableCell>{t('reportMaximumSpeed')}</TableCell>
-              <TableCell>{t('reportDuration')}</TableCell>
-            </TableRow>
-          </TableHead>
-          <TableBody>
-            {items.map((item) => (
-              <TableRow key={item.id}>
-                <TableCell>{formatDate(item.startTime)}</TableCell>
-                <TableCell>{formatDistance(item.startOdometer, distanceUnit)}</TableCell>
-                <TableCell>{formatDate(item.endTime)}</TableCell>
-                <TableCell>{formatDistance(item.endOdometer, distanceUnit)}</TableCell>
-                <TableCell>{formatDistance(item.distance, distanceUnit)}</TableCell>
-                <TableCell>{formatSpeed(item.averageSpeed, speedUnit)}</TableCell>
-                <TableCell>{formatSpeed(item.maxSpeed, speedUnit)}</TableCell>
-                <TableCell>{formatHours(item.duration)}</TableCell>                             
-              </TableRow>
-            ))}
-          </TableBody>
-        </Table>
-      </TableContainer>
+      <DataGrid
+        rows={items} 
+        columns={columns} 
+        hideFooter 
+        autoHeight
+        getRowId={() => Math.random()} />
```

Comment:
Why not just use index for id?

Reply:
Sorry, I didn't understand which index. 

I think we can pass row object to getRowId callback but row doesn't have index.

Reply:
We cannot get item index in the list?

Reply:
Tried witout success.